pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/BaseSegmentUploadConsistentPushIntegrationTest.java

```diff
@@ -0,0 +1,172 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.integration.tests;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.collect.Lists;
+import java.io.File;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import org.apache.commons.io.FileUtils;
+import org.apache.pinot.controller.helix.ControllerTest;
+import org.apache.pinot.plugin.ingestion.batch.common.BaseSegmentPushJobRunner;
+import org.apache.pinot.spi.config.table.TableConfig;
+import org.apache.pinot.spi.config.table.TableType;
+import org.apache.pinot.spi.config.table.ingestion.BatchIngestionConfig;
+import org.apache.pinot.spi.config.table.ingestion.IngestionConfig;
+import org.apache.pinot.spi.data.Schema;
+import org.apache.pinot.spi.ingestion.batch.spec.PinotClusterSpec;
+import org.apache.pinot.spi.ingestion.batch.spec.PinotFSSpec;
+import org.apache.pinot.spi.ingestion.batch.spec.PushJobSpec;
+import org.apache.pinot.spi.ingestion.batch.spec.SegmentGenerationJobSpec;
+import org.apache.pinot.spi.ingestion.batch.spec.TableSpec;
+import org.apache.pinot.spi.utils.builder.ControllerRequestURLBuilder;
+import org.testng.Assert;
+
+
+public abstract class BaseSegmentUploadConsistentPushIntegrationTest extends SegmentUploadIntegrationTest {
```

Comment:
Can we add the extra test function within `SegmentUploadIntegrationTest` to cover this?

Let's do the following:

- Cover the standalone jobs `SegmentTarPushJobRunner, SegmentTarPushJobRunner,SegmentMetadataPushJobRunner` from `SegmentUploadIntegrationTest`
- For platform-specific push job runners (spark, hadoop etc), we can do one of the followings:
  1. We can make sure that our refactored code would run exactly the same code logic and add `TODO` + create the OSS issue for asking help to add the testing for platform-specific push job runners.
  2. If you would like to add the tests, we should do it under each platform specific package under `pinot-batch-ingestion/pinot-batch-ingestion-<platform>`